packages/web-client/app/components/workflow-check/index.ts

```diff
@@ -0,0 +1,20 @@
+import Component from '@glimmer/component';
+
+interface WorkflowCheckArgs {
+  onComplete: () => void;
+  check: () => Promise<boolean>;
+  cancel: () => void;
+}
+
+export default class WorkflowCheck extends Component<WorkflowCheckArgs> {
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    this.args.check().then((v) => {
```

Comment:
Tiny DX suggestion: maybe `successful` or something would be a good name for this callback variable?